### PR TITLE
refactor!(airbyte-cdk): remove availability strategy

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolving https://github.com/airbytehq/airbyte-internal-issues/issues/8521

Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2150,  https://github.com/airbytehq/airbyte-internal-issues/issues/2346

### Phase 1 (this PR)
- remove call to availability strategy from abstract source `read()`
- mark availabilitystrategy as `deprecated`

DOD: `availabilitystrategy` and its interfaces are still presented in codebase, but not called by CDK during the read.

### Phase 2

completely remove any mentions about `availabilitystrategy`

## Review guide

1. `airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py`

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by maxi297 — COMMENTED** [review]
Should we close issues like:
* https://github.com/airbytehq/airbyte-internal-issues/issues/2150
* https://github.com/airbytehq/airbyte-internal-issues/issues/2346

Also, I was checking this change and I fear it might affect a lot of mock server tests as when using the annotation, we are validating that the mocked requests are actually called (see [this](https://github.com/airbytehq/airbyte/blob/100f4e0f34e98dbc540790a9d836f511cfd4aa51/airbyte-cdk/python/airbyte_cdk/test/mock_http/mocker.py#L118-L124)). 

Using the annotation was the intended way at first to use the HttpMocker but some have moved away from that in favor of [this for example](https://github.com/airbytehq/airbyte/blob/22e36de5c2b0a394651c1e54f12a6b58dd70883b/airbyte-integrations/connectors/source-shopify/unit_tests/integration/test_bulk_stream.py#L52-L60) so it might not be too bad. 

In terms of code, this LGTM but I remember @brianjlai mentioning some prerequisite in other to implement this so I would like him to share and validate those. I'm good to approve after that!

**Comment by artem1205** [comment]
> Should we close issues like: ...

update description to close this issue once PR will be merged.

> I fear it might affect a lot of mock server as when using the annotation, we are validating that the mocked requests are actually called 

yes, integration tests should be updated after that removing 1 _extra_ (initial) request should be enough.

**Comment by artem1205** [comment]
Failed Zendesk Support unit test is expected:: Need to remove extra call from httpmocker for availabilitystrategy.

**Comment by maxi297** [comment]
Regarding scope: Should we do something about the availability strategy in the concurrent cdk?

**Comment by brianjlai** [comment]
> In terms of code, this LGTM but I remember @brianjlai mentioning some prerequisite in other to implement this so I would like him to share and validate those. I'm good to approve after that!

One of the potential prereqs was that we have a way to override the default error mappings for a connector that might return permissions errors as a 400 for example. But I believe we have that supported in HttpClient / HttpStatusErrorHandler.

https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/http_status_error_handler.py#L20

I think failing integration tests is simply something we'll have to deal with, but let's at least notify the team. I don't think we need to expand the scope so drastically to fixing all in one go. Just as long as we're aware that many integration tests are affected since we'll have at least one last call :/ unless anyone has better ideas

Will ✅ once the conflicts are updated

**Review by bazarnov — APPROVED** [review]

**Comment by artem1205** [comment]
@maxi297 ,
`Availability strategy` removed from concurrent CDK in https://github.com/airbytehq/airbyte/pull/40682/commits/f5540b7f316f1141ce6f433ebfeff57e7a0f802d.
`INCOMPLETE` status is handled in `def _on_stream_is_done` https://github.com/airbytehq/airbyte/blob/48af92ad788fb68c753d9bc114328fb973c82a46/airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py#L213-L216

**Review by brianjlai — APPROVED** [review]
excited to finally see availability strategy go away!

I think something I'm trying to understand is this sounds like Phase 1. But the `HttpAvailabilityStrategy` is still used across a few different places in the CDK like the low-code framework in `check_stream.py` and in the downstream sources. Is the final goal to fully deprecate the `AvailabilityStrategy`. And what are the phases to get there?

I don't need to block on this question, but understanding the steps will help with reviewing so I know the incremental steps to reach the final goal. I was a bit confused by this PR at first because availability strategy was still being used in a number of places

File: airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source.py
Comment (artem1205) [reply airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source.py:84]: agreed, deleted

File: airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py
Comment (artem1205) [reply airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py]: 👍 updated

File: airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py
Comment (artem1205) [reply airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py:131]: You're right. This is planned to be in the Phase 2.

I've Updated PR description for better understanding.

**Pull request merged**
